metadata/src/main/java/org/apache/kafka/image/MetadataImage.java

```diff
@@ -120,10 +121,16 @@ public AclsImage acls() {
     }
 
     public void write(Consumer<List<ApiMessageAndVersion>> out) {
+        // We use the minimum KRaft metadata version if this image does
+        // not have a specific version set.
+        MetadataVersion metadataVersion = features.metadataVersion();
+        if (metadataVersion.equals(MetadataVersion.UNINITIALIZED)) {
+            metadataVersion = MetadataVersion.IBP_3_0_IV1;
+        }
```

Comment:
I think we need a similar change for the `ClusterControlManager` iterator. If I am not mistaken it is possible for the `FeatureControlManager` to return `UNINITIALIZED` in both active controllers and inactive controllers.

Should we document this default version in `MetadataVersion` instead?

cc @mumrah

Reply:
@mumrah is removing `UNINITIALIZED` in https://github.com/apache/kafka/pull/12250. We can remove this logic afterwards.